federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -332,13 +338,45 @@ func (nc *NamespaceController) delete(namespace *api_v1.Namespace) {
 		nc.eventRecorder.Event(namespace, api.EventTypeNormal, "DeleteNamespace", fmt.Sprintf("Marking for deletion"))
 		_, err := nc.federatedApiClient.Core().Namespaces().Update(updatedNamespace)
 		if err != nil {
-			glog.Errorf("Failed to update namespace %s: %v", updatedNamespace.Name, err)
-			nc.deliverNamespace(namespace.Name, 0, true)
-			return
+			return fmt.Errorf("failed to update namespace: %v", err)
 		}
 	}
 
-	// TODO: delete all namespace content.
+	// Right now there is just 3 types of objects: ReplicaSet, Secret and Service.
+	// Temprairly these items are simply deleted one by one to squeeze this code into 1.4.
+	// TODO: Make it generic (like in the regular namespace controller) and parallel.
+	serviceList, err := nc.federatedApiClient.Core().Services(namespace.Name).List(api.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get service list: %v", err)
+	}
+	for _, item := range serviceList.Items {
+		err = nc.federatedApiClient.Core().Services(namespace.Name).Delete(item.Name, &api.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to delete service %s: %v", item.Name, err)
+		}
+	}
+
```

Comment:
Yes it does: https://github.com/kubernetes/kubernetes/blob/1c5af3e72feb00cf6510c8c10abb991a0895e620/federation/client/clientset_generated/federation_release_1_4/typed/core/v1/service.go#L37